course_content/notebooks/dask_intro.ipynb

```diff
@@ -229,27 +225,217 @@
+    "from time import sleep\n",
+    "\n",
+    "@delayed(pure=True)\n",
+    "def add(a, b):\n",
+    "    return a + b\n",
+    "\n",
+    "@delayed(pure=True)\n",
+    "def mul(a, b):\n",
+    "    return a * b\n",
+    "\n",
+    "@delayed(pure=True)\n",
+    "def inc(a):\n",
+    "    return a + 1"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "By running the cells below you will see **z** is a delayed object and the dask methods **visualize()** and **compute()** can be used on them like any other dask data type."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "z = add(1, 2)\n",
+    "z"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "z.visualize()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "z.compute()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Similarly **c** is a delayed object which can be visualized and computed; it is slightly more complex than x on its own."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "a = inc(1)\n",
+    "b = mul(1, 2)\n",
+    "c = add(a, b)\n",
+    "c"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "c.visualize()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "c.compute()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "It is worth noting that here we have created objects which are \"lazy\"; when they are created they are not immediately executed. Their execution is **delayed** until a time which we the users (or a scheduler) determine."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Exercise\n",
+    "Run the code in the cell below and then visualize and compute **total**."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "results = []\n",
+    "for x in range(4):\n",
+    "    a = inc(1)\n",
+    "    b = mul(1, x)\n",
+    "    c = add(a, b)\n",
+    "    results.append(c)\n",
+    "\n",
+    "total = delayed(sum, pure=True)(results)\n",
+    "total"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Visualize total here."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Compute total here."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Solutions"
```

Comment:
For consistency with the other courses, it might make more sense to put the solutions in a separate dir: 
https://github.com/SciTools/courses/tree/master/course_content/solutions